libpod/container_internal_linux.go

```diff
@@ -466,30 +466,53 @@ func (c *Container) setupSystemd(mounts []spec.Mount, g generate.Generator) erro
 		g.AddMount(tmpfsMnt)
 	}
 
-	// rootless containers have no write access to /sys/fs/cgroup, so don't
-	// add any mount into the container.
-	if !rootless.IsRootless() {
+	unified, err := cgroups.IsCgroup2UnifiedMode()
+	if err != nil {
+		return err
+	}
+
+	if unified {
 		cgroupPath, err := c.CGroupPath()
 		if err != nil {
 			return err
 		}
-		sourcePath := filepath.Join("/sys/fs/cgroup/systemd", cgroupPath)
-
+		sourcePath := filepath.Join("/sys/fs/cgroup", cgroupPath)
```

Comment:
I wonder if we should be checking if a mount already exists at `/sys/fs/cgroup` and not doing this if so? We might be overriding a user mount.

Reply:
I've tried and at this time we have already added the default `/sys/fs/cgroup` so it wouldn't work.

Reply:
Ahhh. Alright, that makes sense.